app/scripts/modules/core/src/managed/EnvironmentsList.tsx

```diff
@@ -1,16 +1,61 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { flatMap, uniq } from 'lodash';
 
 import { IManagedEnviromentSummary, IManagedResourceSummary, IManagedArtifactSummary } from '../domain';
 
 import { NoticeCard } from './NoticeCard';
 import { ManagedResourceObject } from './ManagedResourceObject';
 import { EnvironmentRow } from './EnvironmentRow';
+import { AccountService } from 'core/account';
 
 function shouldDisplayResource(resource: IManagedResourceSummary) {
   //TODO: naively filter on presence of moniker but how should we really decide what to display?
   return !!resource.moniker;
 }
 
+const getAccountNamesFromResources = (
+  resources: string[],
+  resourcesById: { [id: string]: IManagedResourceSummary },
+): string[] => {
+  return uniq(flatMap(resources, resource => resourcesById[resource]?.locations?.account).filter(Boolean));
+};
+
+interface IEnvironmentProps {
+  environment: IManagedEnviromentSummary;
+  resourcesById: { [id: string]: IManagedResourceSummary };
+}
+
+function Environment({ environment, resourcesById }: IEnvironmentProps) {
```

Comment:
I don't have a clear idea on how often this is going to get rendered between data refreshes at the moment. I'm going to wrap it for now to avoid `isProd` calculations.